### ISSUE TITLE
The auto value of the version setting in the DOS section does not emulate the latest MS-DOS versions

### ISSUE THREAD

**Issue by BridgeHeadland** [joncampbell123/dosbox-x#3998]
### Describe the bug

At least the value does not emulate MS-DOS 7.1 and 8.0, as far as I know. I don't know if it's a bug or if it was done on purpose, but I doubt it was done on purpose.

### Steps to reproduce the behaviour

Set the version setting in the DOS section to auto and run an executable file that cannot be run in MS-DOS 7.0 or earlier.

### Expected behavior

I expect the auto value to emulate all DOS versions (also DOSs not developed by Microsoft) without problems, not least MS-DOS 7.1 and 8.0.

### What operating system(s) this bug have occurred on?

Windows 10 Home, 21H2, 19044.2364

### What version(s) of DOSBox-X have this bug?

At least the latest ones, but I assume all the versions with auto value to the version setting have this bug - if it is a bug.

### Have you checked that no similar bug report(s) exist?

- [X] I have searched and didn't find any similar bug report.

### Code of Conduct & Contributing Guidelines

- [X] I agree to follow the code of conduct and the contributing guidelines.

**Comment by rderooy** [comment]
are you talking about this?

```ini
[dos]
#                                              ver: Set DOS version. Specify as major.minor format. A single number is treated as the major version (compatible with LFN support). Common settings are:
#                                                     auto (or unset)                  Pick a DOS kernel version automatically
#                                                     3.3                              MS-DOS 3.3 emulation (not tested!)
#                                                     5.0                              MS-DOS 5.0 emulation (recommended for DOS gaming)
#                                                     6.22                             MS-DOS 6.22 emulation
#                                                     7.0                              MS-DOS 7.0 (or Windows 95 pure DOS mode) emulation
#                                                     7.1                              MS-DOS 7.1 (or Windows 98 pure DOS mode) emulation
#                                                     Long filename (LFN) support will be enabled with a reported DOS version of 7.0 or higher with "lfn=auto" (default).
#                                                     Similarly, FAT32 disk images will be supported with a reported DOS version of 7.1 or higher.
ver                                              =
```

It can be somewhat misleading. But if you leave it blank, it will choose DOS version 5.0

**Comment by Torinde** [comment]
Agree that it's useful to spell out that auto/unset = 5.0. 
@BridgeHeadland, auto doesn't mean "automatically adjusting depending on the applications user runs" (that I think is difficult to implement - unless you have an idea how DOSbox can analyze/detect what DOS version the application requires).

"I expect the auto value to emulate all DOS versions (also DOSs not developed by Microsoft)" - is there a specific feature/distinction that you would like to be enabled/disabled (e.g. you want PC DOS 6.3 in the selector, because it differs from 6.22 in what way?)

**Comment by maron2000** [comment]
I think we should not use ver 6.3 because Microsoft version does not exist.
If we mix up MS and PC DOS versions, there are no benefit but only confusion for 7.0 and after.

**Comment by rderooy** [comment]
DOSBox-X generally uses the MS versions, not the IBM versions. e.g. IBM PC DOS 7.0/2000 does not have VFAT (LFN), while the MS DOS 7.0 (aka Win95 DOS) does.

Prior to DOS 6 there was no real difference between MS and IBM versions. With DOS 6 they started to diverge.
https://dosbox-x.com/wiki/Guide%3ADOS-Installation-in-DOSBox%E2%80%90X#_dos_version_highlights

**Comment by rderooy** [comment]
@Torinde Generally speaking, setting `ver=` will not have any effect in DOSBox-X emulation. It only effects external programs or utilities that you choose to run.

The exception to this is if you set ver to >=7.0 which will enable by default LFN support, and setting >= 7.1 will enable FAT32 emulation.

You can indeed say `ver=6.3`, but again, it will make no difference to DOSBox-X emulation. In practice, unless you want to run utilities shipped with IBM PC DOS 6.3, it is pointless.

e.g. setting `ver=1.0` will not suddenly disable the ability to create directories.
